src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -53,7 +53,7 @@ inline void asum(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<
             auto res_ = sc.get_mem<cuDataType2 *>(res_acc);
             cublasStatus_t err;
             // ASUM does not support negative index
-            CUBLAS_ERROR_FUNC(func, err, handle, n, x_, std::abs(incx), res_);
+            CUBLAS_ERROR_FUNC_T(func_name, func, err, handle, n, x_, std::abs(incx), res_);
```

Comment:
```suggestion
            CUBLAS_ERROR_FUNC_T("asum", func, err, handle, n, x_, std::abs(incx), res_);
```
I think passing the function name this way is simpler and will require less change to code / function signatures. It also aligns with what we do when we `throw unimplemented` in some other cases.

Reply:
Yes, but the intent of ``CUBLAS_ERROR_FUNC``/``_T`` is throw an exception in case of a cublas function error, so its message should contain the cublas function name.

Reply:
I agree, using cublas function name in the final message is more useful for the user. However, I think, we haven't finalized/implemented how exceptions are going to be implemented at oneMKL interfaces layer. What I mean is, it is not clear to me how we are going to pass the exceptions coming from lower backends. For this PR, I am ok with passing cublas function names, however, depending on overall strategy, we would change it I guess.